diff -ur fritz/src/tools.h fritz.new/src/tools.h
--- fritz/src/tools.h	2005-07-07 00:00:00.000000000 +0200
+++ fritz.new/src/tools.h	2006-01-26 17:29:53.000000000 +0100
@@ -71,13 +71,14 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
+#ifndef atomic_xchg
 static inline unsigned long atomic_xchg (
 	volatile atomic_t *	v, 
 	unsigned		value
 ) {
 	return __xchg (value, &v->counter, sizeof (unsigned));
 } /* atomic_xchg */
-
+#endif
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 #ifndef NDEBUG
