diff -urp fritz/src.org/driver.c fritz/src/driver.c
--- fritz/src.org/driver.c	2007-09-29 18:44:31.000000000 +0000
+++ fritz/src/driver.c	2007-09-30 12:13:38.000000000 +0000
@@ -107,7 +107,7 @@ static int			card_id			= 0;
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 static void scheduler (unsigned long data);
-static irqreturn_t irq_handler (int irq, void * args, struct pt_regs * regs);
+static irqreturn_t irq_handler (int irq, void * args);
 
 static DECLARE_TASKLET_DISABLED (scheduler_tasklet, scheduler, 0);
 
@@ -374,9 +374,9 @@ static int install_card (card_t * card) 
 			card->irq, 
 			&irq_handler, 
 #if defined (__fcpci__) || defined (__fcpcmcia__)
-			SA_INTERRUPT | SA_SHIRQ, 
+			IRQF_DISABLED | IRQF_SHARED, 
 #else
-			SA_INTERRUPT, 
+			IRQF_DISABLED, 
 #endif
 			TARGET, 
 			card
@@ -836,11 +836,10 @@ static void scheduler (unsigned long dat
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-static irqreturn_t irq_handler (int irq, void * args, struct pt_regs * regs) {
+static irqreturn_t irq_handler (int irq, void * args) {
 	int	res	= IRQ_NONE;
 	
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	if (args != NULL) {
 		assert (capi_lib->cm_handle_events != NULL);
 		if (atomic_read (&scheduler_id) == smp_processor_id ()) {
