--- fritz/src/driver.c.orig	2009-11-28 12:03:12.000000000 +0100
+++ fritz/src/driver.c	2009-11-28 12:12:50.936005814 +0100
@@ -194,7 +194,7 @@
 	}
 	lib_strncpy (ctrl->serial, tmp, CAPI_SERIAL_LEN);
 	lib_memcpy (&ctrl->profile, card->string[6], sizeof (capi_profile));
-	strncpy (ctrl->manu, "AVM GmbH", CAPI_MANUFACTURER_LEN);
+	lib_strncpy (ctrl->manu, "AVM GmbH", CAPI_MANUFACTURER_LEN);
 	ctrl->version.majorversion = 2;
 	ctrl->version.minorversion = 0;
 	tmp = card->string[0];
