diff -ur fritz/src.orig/tools.h fritz/src/tools.h
--- fritz/src.orig/tools.h	2009-09-29 01:07:39.212061769 +0200
+++ fritz/src/tools.h	2009-09-29 01:00:18.000000000 +0200
@@ -71,14 +71,14 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-#ifndef atomic_xchg
-static inline unsigned long atomic_xchg (
-	volatile atomic_t *	v, 
-	unsigned		value
-) {
-	return __xchg (value, &v->counter, sizeof (unsigned));
-} /* atomic_xchg */
-#endif
+//#ifndef atomic_xchg
+//static inline unsigned long atomic_xchg (
+//	volatile atomic_t *	v, 
+//	unsigned		value
+//) {
+//	return __xchg (value, &v->counter, sizeof (unsigned));
+//} /* atomic_xchg */
+//#endif
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 #ifndef NDEBUG
